### PR TITLE
feat: native / all archs

### PR DESCRIPTION
**Pull request by henryiii**
I rather forget we already had the correct definition of "auto", but this adds the other two as mentioned in #528.

### CODE REVIEW

File: cibuildwheel/util.py [hunk 1]
```diff
@@ -184,6 +188,17 @@ def auto_archs(platform: PlatformName) -> 'Set[Architecture]':
             result.add(Architecture.x86)
         return result
 
+    @staticmethod
+    def all_archs(platform: PlatformName) -> 'Set[Architecture]':
```
Comment (YannickJadoul) [inline cibuildwheel/util.py:192]: Does this overlap with `ALLOWED_ARCHITECTURES`? In values, definitely. But conceptually as well?

Comment (henryiii) [reply cibuildwheel/util.py:192]: Yes, they should be the same. Good catch!

File: cibuildwheel/util.py [hunk 2]
```diff
@@ -255,19 +270,13 @@ def detect_ci_provider() -> Optional[CIProvider]:
...
 
-ALLOWED_ARCHITECTURES = {
-    'linux': {Architecture.x86_64, Architecture.i686, Architecture.aarch64, Architecture.ppc64le, Architecture.s390x},
-    'macos': {Architecture.x86_64},
-    'windows': {Architecture.AMD64, Architecture.x86},
-}
-
 
 def allowed_architectures_check(
     name: PlatformName,
```
Comment (joerick) [inline cibuildwheel/util.py:266]: Might have missed this first time around, but generally this variable should be called 'platform'

**Review by joerick — COMMENTED** [review]
Looks good! I guess we just need these terms adding to the CIBW_ARCHS_LINUX docs?

**Comment by henryiii** [comment]
:eyes:?

**Review by joerick — APPROVED** [review]
Great!

**Pull request merged**